app.js

```diff
@@ -1,37 +1,54 @@
 const dayjs = require("dayjs");
 const EKSToken = require("aws-eks-token");
+const AWS = require("aws-sdk");
+const { promisify } = require("util");
 const { listRegions } = require("./autocomplete");
+const { CLUSTER_NAME_NOT_PROVIDED_MESSAGE, EXPIRES_INVALID_MESSAGE } = require("./consts");
+const parsers = require("./parsers");
 
-async function getToken(action, settings) {
-  if (!action.params.clusterName) {
-    throw new Error("not given cluster name");
+async function getToken({ params }, settings) {
+  const clusterName = parsers.string(params.clusterName);
+  const region = parsers.autocomplete(params.region);
+  const expires = parsers.number(params.expires) || 60;
+  const secretAccessKey = parsers.string(params.secretAccessKey || settings.secretAccessKey);
+  const accessKeyId = parsers.string(params.accessKeyId || settings.accessKeyId);
+
+  if (!clusterName) {
+    throw new Error(CLUSTER_NAME_NOT_PROVIDED_MESSAGE);
   }
-  const expires = action.params.expires || "60";
   if (!Number.isInteger(parseFloat(expires))) {
-    throw new Error("expires must be an integer");
+    throw new Error(EXPIRES_INVALID_MESSAGE);
   }
 
+  const credentials = { secretAccessKey, accessKeyId };
+
+  const EKS = new AWS.EKS({ credentials, region });
+
   EKSToken.config = {
-    accessKeyId: action.params.accessKeyId || settings.accessKeyId,
-    secretAccessKey: action.params.secretAccessKey || settings.secretAccessKey,
-    region: action.params.region,
+    ...credentials,
+    region,
   };
 
   const reqTime = dayjs();
+  const dateFormat = "YYYY-MM-DDTHH:mm:ss[Z]";
   const token = await EKSToken.renew(
-    action.params.clusterName,
+    clusterName,
     expires,
-    reqTime.utc().format("YYYYMMDDTHHmmss[Z]"),
+    reqTime.utc().format(dateFormat),
   );
-  const expirationTime = reqTime.add(parseInt(expires, 10), "s").utc().format("YYYY-MM-DDTHH:mm:ss[Z]");
+  const {
+    cluster,
+  } = await promisify(EKS.describeCluster.bind(EKS))({ name: clusterName });
```

Comment:
nice try, but all aws methods provide built-in promise support via .promise() method.
try `... = await EKS.describeCluster({name: clusterName}).promise()` instead :)